### DISCUSSION TITLE
Is there a preferred way to extract symbols from expression in the correct order?

### DISCUSSION THREAD

**Discussion by kgoebler** [sympy/sympy#25498] [General]
The following problem (in simplified terms) has been bugging me a bit.

The `args` to pass to `lambdify` have to match exactly the order sympy stores expressions internally (to my best knowledge).
I've not found a simple way to extract free symbols in the correct internally saved order from an expression to store in a list an pass to 
`lambdify`. 

In this simple example below, one might expect this should work and return `0`. 

Currently I have a workaround by extracting elements from  `my_expr._sorted_args` (potentially in several repeating steps) but it's ugly and not very practical. 

Is there a better way of doing this?

Thanks in advance!

```python
  import numpy as np
  from sympy import symbols, Symbol, lambdify, free_symbols, srepr
  
  x, y, z = symbols("x,y,z")
  
  my_expr = x * (2 * y - z)
  args = list(my_expr.atoms(Symbol))
  # args = list(my_expr.free_symbols) same unordered and unpredictable behavior
  
  # These give the correct order, but don't return only symbols (messy to extract symbols)
  # srepr(my_expr)
  # my_expr._sorted_args
  
  
  f = lambdify(args=args, expr=my_expr)
  
  f(np.array([-2]), np.array([1]), np.array([2]))
  # returns array([8]) rather than array([0])
```

**Comment by moorepants** [comment]
Here is one way:

```
ordered(expression.free_symbols)
```

**Reply by kgoebler** [reply]
Oh wow never thought of trying this :-) thanks a lot for the quick reply and fix!
